pyproject.toml

```diff
@@ -1,3 +1,111 @@
+[build-system]
+build-backend = "hatchling.build"
+requires = ["hatchling"]
+
+[project]
+name = "basket-client"
+dynamic = ["version"]
+description = "A Python client for Mozilla's basket service."
+readme = "README.rst"
+requires-python = ">=3.7"
+license = {text = "BSD-3-Clause"}
+keywords = ["mozilla", "basket"]
+authors = [
+  { name = "Michael Kelly" },
+]
+maintainers = [
+  { name = "Paul McLanahan" },
+  { name = "Rob Hudson" },
+]
+classifiers = [
+  "Development Status :: 5 - Production/Stable",
+  "Environment :: Web Environment",
+  "Intended Audience :: Developers",
+  "License :: OSI Approved :: BSD License",
+  "Natural Language :: English",
+  "Programming Language :: Python",
+  "Programming Language :: Python :: 3.7",
+  "Programming Language :: Python :: 3.8",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
+  "Programming Language :: Python :: 3.11",
+  "Topic :: Communications",
+  "Topic :: Software Development :: Libraries",
+]
+dependencies = [
+  "requests",
```

Comment:
I'd restrict this to the current major version. That's the most flexible scenario and protects against big API changes in requests breaking everything assuming they do versioning well.

```
"requests<3.0.0",
```

The thinking here is that you're protecting against CI or deploys auto-installing a very different version than what you've made sure works.